nav2_navfn_planner/src/navfn_planner.cpp

```diff
@@ -284,6 +288,24 @@ NavfnPlanner::makePlan(
     // extract the plan
     if (getPlanFromPotential(best_pose, plan)) {
       smoothApproachToGoal(best_pose, plan);
+
+      size_t plan_size = plan.poses.size();
+      if (use_final_approach_orientation_ && plan_size > 1) {
```

Comment:
This would have to be > 2 in order to do the approach_pose.

What would you do if you had a path that came back with only 1 or 2 poses? It would then return the actual angle instead making your TEB planner backout to hit that orientation goal.

Reply:
That will happen, and happen often, with any reasonably large scaled deployment.

Reply:
This is working for a path with only 2 poses (plan_size = 2)
However I am not sure to see what should happen for a path with one pose only. We probably don't want the robot to move at all if `use_final_approach_orientation=true` (i.e not rotate if it doesn't have to change its x,y position), hence we need our only pose to have the same orientation as the current robot orientation. Hence the start pose orientation. Does it make sense ?

Reply:
yes,  the starting pose orientation makes sense, we should cover that edge case